Reflect in the documentation that the LOHThreshold has to be less than region_size

cshung opened the pull request:
## Summary

We recently discovered that we cannot let the user to allocate arbitrarily large objects in the small object heap, therefore we implement a change (https://github.com/dotnet/runtime/pull/95485) in the runtime and this change in the doc reflects that.

The actual cap is the `gc_region_size`. It is an implementation detail and it changes according to different user/machine configurations. Therefore, instead of specifying the maximum, we just said it could be capped. I am open to suggestion on how to make it sounds more clear.

The product code change fixes https://github.com/dotnet/runtime/issues/95219, and we don't have a docs issue for this.

@Maoni0, @dotnet/gc 

<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/runtime-config/garbage-collector.md](https://github.com/dotnet/docs/blob/a90b3470f6baf64a5278d1b68ce45628ed7d365a/docs/core/runtime-config/garbage-collector.md) | [Runtime configuration options for garbage collection](https://review.learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector?branch=pr-en-us-38763) |


<!-- PREVIEW-TABLE-END -->

gewarren commented on docs/core/runtime-config/garbage-collector.md:
```diff
@@ -669,6 +669,7 @@ Project file:
 - Specifies the threshold size, in bytes, that causes objects to go on the large object heap (LOH).
 - The default threshold is 85,000 bytes.
 - The value you specify must be larger than the default threshold.
+- The value can be capped by the runtime to the maximum possible size for the current configuration.
```
Should we add that this is in .NET 9 and later, or is that unnecessary?

cshung replied:
I think it is unnecessary. In principle, the value should have been capped in .net 7/8 as well, we just don't have a business reason to fix those earlier versions, yet. We might if we find a reason to.

Maoni0 commented:
it'd be nice to mention how the user can call the config api to find out the actual value that's set.

cshung committed:
Reflect in the documentation that the LOHThreshold has to be less than region_size

We recently discovered that we cannot let the user to allocate arbitrarily large objects in the small object heap, therefore we implement a change (https://github.com/dotnet/runtime/pull/95485) in the runtime and this change in the doc reflects that.

The actual cap is the `gc_region_size`. It is an implementation detail and it changes according to different user/machine configurations. Therefore, instead of specifying the maximum, we just said it could be capped.

cshung committed:
Describe how to inspect the value through the GC.GetConfigurationVariables API

gewarren commented on docs/core/runtime-config/garbage-collector.md:
```diff
@@ -669,6 +669,7 @@ Project file:
 - Specifies the threshold size, in bytes, that causes objects to go on the large object heap (LOH).
 - The default threshold is 85,000 bytes.
 - The value you specify must be larger than the default threshold.
+- The value can be capped by the runtime to the maximum possible size for the current configuration. The value actually being used at runtime can be inspected through the [GC.GetConfigurationVariables](https://learn.microsoft.com/en-us/dotnet/api/system.gc.getconfigurationvariables) API.
```
```suggestion
- The value might be capped by the runtime to the maximum possible size for the current configuration. You can inspect the value in use at run time through the <xref:System.GC.GetConfigurationVariables?displayProperty=nameWithType> API.
```

cshung committed:
Update docs/core/runtime-config/garbage-collector.md

Co-authored-by: Genevieve Warren <24882762+gewarren@users.noreply.github.com>
